projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/class-wpcom-admin-bar.php

```diff
@@ -60,6 +60,11 @@ class WPCOM_Admin_Bar extends \WP_Admin_Bar {
 	 * }
 	 */
 	public function add_node( $args ) {
+		// Ensure $args is an array.
+		if ( is_object( $args ) ) {
```

Comment:
I think the most robust way is to just call `parent::add_node( $args );` and bail, if `$args` is not an array. Similar to the next if-block at line 68.

(It means we only care to manipulate the nodes on wpcom side only if `$args` is array)

Reply:
Yeah, I like that. 👍 I updated the PR.

Reply:
yep, it looks better to return early if it's not an array because the args might be a string as well 😓

```
// Shim for old method signature: add_node( $parent_id, $menu_obj, $args ).
if ( func_num_args() >= 3 && is_string( $args ) ) {
	$args = array_merge( array( 'parent' => $args ), func_get_arg( 2 ) );
}

if ( is_object( $args ) ) {
	$args = get_object_vars( $args );
}
```